specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2023-03-01/common-types.json

```diff
@@ -0,0 +1,232 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-03-01",
+    "title": "CostManagementClient"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.CostManagement/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "operationId": "Operations_List",
+        "description": "Lists all of the available cost management REST API operations.",
+        "x-ms-examples": {
+          "OperationList": {
+            "$ref": "./examples/OperationList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
+      "description": "Error response indicates that the service is not able to process the incoming request. The reason is provided in the error message. \n\nSome Error responses: \n\n * 429 TooManyRequests - Request is throttled. Retry after waiting for the time specified in the \"x-ms-ratelimit-microsoft.consumption-retry-after\" header. \n\n * 503 ServiceUnavailable - Service is temporarily unavailable. Retry after waiting for the time specified in the \"Retry-After\" header.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The details of the error.",
+          "$ref": "#/definitions/ErrorDetails"
+        }
+      }
+    },
+    "ErrorDetails": {
+      "description": "The details of the error.",
+      "type": "object",
+      "properties": {
+        "code": {
```

Comment:
>code [](http://example.com/codeflow?start=9&length=4)

this must be a string [[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content)](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content)

Reply:
This will be a breaking change for all other APIs and an inconsistency if introduced for just the new API as mentioned offline in teams. Must we change all APIs?

Reply:
This was a string in your previous API version and a string is the official error contract in Azure.  Doing anything different here opens you up to all sorts of potential problems.

Reply:
It is an int in previous api versions as far as I can tell in the main branch and public facing documentation. https://learn.microsoft.com/en-us/rest/api/cost-management/benefit-utilization-summaries/list-by-billing-account-id?tabs=HTTP#errordetails

Reply:
2022-10-01 common-types ErrorDetails.code.type is string.  If you compare the latest iteration of this PR with the base iteration it also shows this type change as a diff in this PR.  

2022-10-01:  https://github.com/Azure/azure-rest-api-specs/blob/eeab098444d6258af140a031c222d8636cd7db4f/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/common-types.json#L88-L92